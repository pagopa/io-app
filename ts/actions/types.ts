/**
 * Defines Flow types for the available actions and store related stuff.
 */

import { NavigationAction } from "react-navigation";
import {
  Dispatch as DispatchAPI,
  MiddlewareAPI as ReduxMiddlewareAPI,
  Store as ReduxStore,
  StoreEnhancer as ReduxStoreEnhancer
} from "redux";

import { GlobalState } from "../reducers/types";
import { ApplicationActions } from "../store/actions/application";
import { APP_STATE_CHANGE_ACTION } from "../store/actions/constants";
import { ErrorActions } from "../store/actions/error";
import { NotificationsActions } from "../store/actions/notifications";
import { OnboardingActions } from "../store/actions/onboarding";
import { ProfileActions } from "../store/actions/profile";
import { SessionActions } from "../store/actions/session";

export type ApplicationState = "background" | "inactive" | "active";

export type ApplicationStateAction = Readonly<{
  type: typeof APP_STATE_CHANGE_ACTION;
  payload: ApplicationState;
}>;

export type Action =
  | ApplicationActions
  | ApplicationStateAction
  | NavigationAction
  | SessionActions
  | OnboardingActions
  | ProfileActions
  | NotificationsActions
  | ErrorActions;

/* eslint-disable no-use-before-define */
// We need to disable the eslint rule because of a problem described here
// @https://github.com/babel/babel-eslint/issues/485
export type GetState = () => GlobalState;

export type Dispatch = DispatchAPI<Action, GlobalState>;

export type Store = ReduxStore<GlobalState, Action>;

export type StoreEnhancer = ReduxStoreEnhancer<GlobalState>;

export type MiddlewareAPI = ReduxMiddlewareAPI<Dispatch, GlobalState>;

// Props injected by react-redux connect() function
export type ReduxProps = Readonly<{
  dispatch: Dispatch;
}>;
