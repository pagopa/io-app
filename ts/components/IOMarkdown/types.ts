import { VSpacer } from "@pagopa/io-app-design-system";
import {
  TxtHeaderNode,
  TxtListNode,
  TxtListItemNode,
  TxtParagraphNode,
  TxtStrNode,
  ASTNodeTypes,
  TxtStrongNode,
  TxtEmphasisNode,
  TxtLinkNode,
  TxtImageNode,
  TxtBlockQuoteNode,
  TxtCodeBlockNode,
  TxtCodeNode,
  TxtBreakNode,
  TxtHtmlNode,
  TxtHorizontalRuleNode,
  AnyTxtNode
} from "@textlint/ast-node-types";
import { ReactNode, ComponentProps } from "react";

export type Renderer = (txtNode: AnyTxtNodeWithSpacer) => ReactNode;
export type RuleRenderer<P, F extends Renderer | void = void> = (
  param: P,
  renderer: F
) => ReactNode;
export type AnyTxtNodeWithSpacer =
  | AnyTxtNode
  | { type: "Spacer"; size: number; key: string };

type TxtNodeMap = {
  [ASTNodeTypes.Header]: TxtHeaderNode;
  [ASTNodeTypes.Paragraph]: TxtParagraphNode;
  [ASTNodeTypes.Str]: TxtStrNode;
  [ASTNodeTypes.Strong]: TxtStrongNode;
  [ASTNodeTypes.Emphasis]: TxtEmphasisNode;
  [ASTNodeTypes.Link]: TxtLinkNode;
  [ASTNodeTypes.List]: TxtListNode;
  [ASTNodeTypes.ListItem]: TxtListItemNode;
  [ASTNodeTypes.Image]: TxtImageNode;
  [ASTNodeTypes.BlockQuote]: TxtBlockQuoteNode;
  [ASTNodeTypes.CodeBlock]: TxtCodeBlockNode;
  [ASTNodeTypes.Code]: TxtCodeNode;
  [ASTNodeTypes.Break]: TxtBreakNode;
  [ASTNodeTypes.Html]: TxtHtmlNode;
  [ASTNodeTypes.HorizontalRule]: TxtHorizontalRuleNode;
};
type RendererMap = {
  [K in keyof TxtNodeMap]: RuleRenderer<TxtNodeMap[K], Renderer>;
} & {
  [ASTNodeTypes.Comment]: RuleRenderer<null>;
  Spacer: RuleRenderer<ComponentProps<typeof VSpacer> & { key: string }>;
};

export type IOMarkdownRenderRules = RendererMap;
