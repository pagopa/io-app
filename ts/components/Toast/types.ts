import { HapticFeedbackTypes } from "react-native-haptic-feedback";
import { IOIcons } from "@pagopa/io-app-design-system";

export type ToastVariant = "neutral" | "error" | "info" | "success" | "warning";

export type Toast = {
  message: string;
  variant?: ToastVariant;
  icon?: IOIcons;
  hapticFeedback?: keyof typeof HapticFeedbackTypes;
};

export type ToastOptions = Omit<Toast, "message">;
