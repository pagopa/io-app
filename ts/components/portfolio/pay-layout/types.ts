/* subtitlesLS and subtitle are mutually exclusive */
type WithSubtitle = {
  hasMainSubtitle: true;
  hasSubtitlesLR: false;
  subtitleText: string;
};

type WithoutSubtitles = {
  hasMainSubtitle: false;
  hasSubtitlesLR: false;
};

type WithSubtitlesLR = {
  hasSubtitlesLR: true;
  hasMainSubtitle: false;
  subtitleLeftText?: string;
  subtitleRightText?: string;
};

type WithTouchable = {
  hasTouchable: true;
  touchableContent: React.ReactElement<any>;
};

type WithoutTouchable = {
  hasTouchable: false;
};

export type TopContentSubtitle = WithSubtitle & WithoutTouchable;
export type TopContentSubtitlesLR = WithSubtitlesLR & WithoutTouchable;
export type TopContentTouchable = WithoutSubtitles & WithTouchable;

export type TopContentSubtitleTouchable = WithSubtitle & WithTouchable;
export type TopContentSubtitlesLRTouchable = WithSubtitlesLR & WithTouchable;
export type TopContentNone = WithoutSubtitles & WithoutTouchable;

export function topContentSubtitle(subtitleText: string): TopContentSubtitle {
  return {
    hasMainSubtitle: true,
    hasSubtitlesLR: false,
    hasTouchable: false,
    subtitleText
  };
}

export function topContentSubtitlesLR(
  subtitleLeftText?: string,
  subtitleRightText?: string
): TopContentSubtitlesLR {
  return {
    hasMainSubtitle: false,
    hasSubtitlesLR: true,
    hasTouchable: false,
    subtitleLeftText,
    subtitleRightText
  };
}

export function topContentTouchable(
  touchableContent: React.ReactElement<any>
): TopContentTouchable {
  return {
    hasMainSubtitle: false,
    hasSubtitlesLR: false,
    hasTouchable: true,
    touchableContent
  };
}

export function topContentSubtitleTouchable(
  touchableContent: React.ReactElement<any>,
  subtitleText: string
): TopContentSubtitleTouchable {
  return {
    hasMainSubtitle: true,
    hasSubtitlesLR: false,
    hasTouchable: true,
    touchableContent,
    subtitleText
  };
}

export function topContentSubtitlesLRTouchable(
  touchableContent: React.ReactElement<any>,
  subtitleLeftText?: string,
  subtitleRightText?: string
): TopContentSubtitlesLRTouchable {
  return {
    hasMainSubtitle: false,
    hasSubtitlesLR: true,
    hasTouchable: true,
    touchableContent,
    subtitleLeftText,
    subtitleRightText
  };
}

export function topContentNone(): TopContentNone {
  return {
    hasMainSubtitle: false,
    hasSubtitlesLR: false,
    hasTouchable: false
  };
}

export type TopContent =
  | TopContentSubtitle
  | TopContentSubtitleTouchable
  | TopContentSubtitlesLR
  | TopContentSubtitlesLRTouchable
  | TopContentTouchable
  | TopContentNone;
