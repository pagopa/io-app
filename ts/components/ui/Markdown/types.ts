/**
 * Types of messages sent from inside the webview.
 */

import * as t from "io-ts";

const LinkMessage = t.type({
  type: t.literal("LINK_MESSAGE"),
  payload: t.type({
    href: t.string
  })
});

const ResizeMessage = t.type({
  type: t.literal("RESIZE_MESSAGE"),
  payload: t.type({
    height: t.number
  })
});

const ScrollEndMessage = t.type({
  type: t.literal("SCROLL_END_MESSAGE")
});

export const WebViewMessage = t.taggedUnion("type", [
  LinkMessage,
  ResizeMessage,
  ScrollEndMessage
]);
