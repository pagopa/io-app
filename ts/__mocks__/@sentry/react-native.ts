export const captureException = jest.fn();
export const captureMessage = jest.fn();
export const captureEvent = jest.fn();
export const setContext = jest.fn();
export const setExtras = jest.fn();
export const setExtra = jest.fn();
export const setTags = jest.fn();
export const setTag = jest.fn();
export const setUser = jest.fn();
export const lastEventId = jest.fn();
export const captureCheckIn = jest.fn();
export const withMonitor = jest.fn();
export const flush = jest.fn().mockResolvedValue(false);
export const close = jest.fn().mockResolvedValue(false);
export const isInitialized = jest.fn().mockReturnValue(false);
export const isEnabled = jest.fn().mockReturnValue(false);
export const addEventProcessor = jest.fn();
export const startSession = jest.fn();
export const endSession = jest.fn();
export const captureSession = jest.fn();
export const ErrorBoundary = jest.fn(({ children }) => children);
