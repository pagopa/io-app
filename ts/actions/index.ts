/**
 * Aggregates all the actions available to components.
 */

export * from './login'
export * from './user'
