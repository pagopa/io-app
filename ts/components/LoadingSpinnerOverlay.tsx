import { Text } from "native-base";
import * as React from "react";
import { StyleSheet } from "react-native";
import I18n from "../i18n";
import variables from "../theme/variables";
import ButtonDefaultOpacity from "./ButtonDefaultOpacity";
import BoxedRefreshIndicator from "./ui/BoxedRefreshIndicator";
import { Overlay } from "./ui/Overlay";

const styles = StyleSheet.create({
  textCaption: {
    padding: variables.contentPadding
  },
  cancelButtonStyle: {
    alignSelf: "center"
  }
});

type Props = Readonly<{
  isLoading: boolean;
  loadingCaption?: string;
  loadingOpacity?: number;
  onCancel?: () => void;
}>;

/**
 * A Component to display and overlay spinner conditionally
 */
class LoadingSpinnerOverlay extends React.Component<Props> {
  public render() {
    const {
      children,
      isLoading,
      loadingCaption,
      loadingOpacity = 0.7,
      onCancel
    } = this.props;
    return (
      <Overlay
        backgroundColor={`rgba(255,255,255,${loadingOpacity})`}
        foreground={
          isLoading && (
            <BoxedRefreshIndicator
              caption={
                loadingCaption && (
                  <Text alignCenter={true} style={styles.textCaption}>
                    {loadingCaption}
                  </Text>
                )
              }
              action={
                onCancel && (
                  <ButtonDefaultOpacity
                    onPress={onCancel}
                    cancel={true}
                    style={styles.cancelButtonStyle}
                  >
                    <Text>{I18n.t("global.buttons.cancel")}</Text>
                  </ButtonDefaultOpacity>
                )
              }
            />
          )
        }
      >
        {children}
      </Overlay>
    );
  }
}

export default LoadingSpinnerOverlay;
