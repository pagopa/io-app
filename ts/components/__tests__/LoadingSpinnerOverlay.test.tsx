import * as React from "react";
import { Text } from "react-native";
import { render } from "@testing-library/react-native";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { appReducer } from "../../store/reducers";
import { applicationChangeState } from "../../store/actions/application";
import { preferencesDesignSystemSetEnabled } from "../../store/actions/persistedPreferences";
import { GlobalState } from "../../store/reducers/types";
import LoadingSpinnerOverlay from "../LoadingSpinnerOverlay";

describe("LoadingSpinnerOverlay", () => {
  it("Should match base no-loading snapshot", () => {
    const component = renderComponent(false);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("Should match base loading snapshot", () => {
    const component = renderComponent(true);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("Should match all-properties and not-loading snapshot", () => {
    const child = <Text>This is a child</Text>;
    const loadingCaption = "This is the loading caption";
    const loadingOpacity = 0.65;
    const onCancelCallback = () => undefined;
    const component = renderComponent(
      false,
      child,
      loadingCaption,
      loadingOpacity,
      onCancelCallback
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("Should match all-properties and loading snapshot", () => {
    const child = <Text>This is a child</Text>;
    const loadingCaption = "This is the loading caption";
    const loadingOpacity = 0.65;
    const onCancelCallback = () => undefined;
    const component = renderComponent(
      true,
      child,
      loadingCaption,
      loadingOpacity,
      onCancelCallback
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

const renderComponent = (
  isLoading: boolean,
  children?: React.ReactNode,
  loadingCaption?: string,
  loadingOpacity?: number,
  onCancel?: () => void
) => {
  const initialState = appReducer(undefined, applicationChangeState("active"));
  const dsState = appReducer(
    initialState,
    preferencesDesignSystemSetEnabled({ isDesignSystemEnabled: true })
  );
  const mockStore = configureMockStore<GlobalState>();
  const store: ReturnType<typeof mockStore> = mockStore(dsState);
  return render(
    <Provider store={store}>
      <LoadingSpinnerOverlay
        isLoading={isLoading}
        loadingCaption={loadingCaption}
        loadingOpacity={loadingOpacity}
        onCancel={onCancel}
      >
        {children}
      </LoadingSpinnerOverlay>
    </Provider>
  );
};
