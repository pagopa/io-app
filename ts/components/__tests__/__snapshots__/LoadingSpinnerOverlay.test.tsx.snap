// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinnerOverlay Should match all-properties and loading snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
  testID="overlayComponent"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "bottom": 0,
          "inset": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1,
        },
        Object {
          "backgroundColor": "rgba(255,255,255,0.65)",
          "opacity": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": 100,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityRole="progressbar"
        accessible={true}
        importantForAccessibility="no-hide-descendants"
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
        testID="LoadingSpinnerTestID"
      >
        <View
          collapsable={false}
          style={
            Object {
              "transform": Array [
                Object {
                  "rotateZ": "0deg",
                },
              ],
            }
          }
          testID="LoadingSpinnerAnimatedTestID"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={24}
            bbWidth={24}
            fill="none"
            focusable={false}
            height={24}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={24}
          >
            <RNSVGGroup
              fill={null}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      736995,
                      1,
                      -16040221,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={0}
                  name="spinner-secondHalf"
                  x1="0%"
                  x2="100%"
                  y1="0%"
                  y2="0%"
                />
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -16040221,
                      1,
                      -16040221,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={0}
                  name="spinner-firstHalf"
                  x1="0%"
                  x2="100%"
                  y1="0%"
                  y2="0%"
                />
              </RNSVGDefs>
              <RNSVGGroup
                propList={
                  Array [
                    "strokeWidth",
                  ]
                }
                strokeWidth={3}
              >
                <RNSVGPath
                  d="M 1.5 12 A 10.5 10.5 0 0 1 22.5 12"
                  propList={
                    Array [
                      "stroke",
                    ]
                  }
                  stroke={
                    Array [
                      1,
                      "spinner-secondHalf",
                    ]
                  }
                />
                <RNSVGPath
                  d="M 22.5 12 A 10.5 10.5 0 0 1 1.5 12"
                  propList={
                    Array [
                      "stroke",
                    ]
                  }
                  stroke={
                    Array [
                      1,
                      "spinner-firstHalf",
                    ]
                  }
                />
                <RNSVGPath
                  d="M 1.5 12 A 10.5 10.5 0 0 1 1.5 11.25"
                  propList={
                    Array [
                      "stroke",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4278927075}
                  strokeLinecap={1}
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
      <View
        style={
          Object {
            "padding": 24,
          }
        }
      >
        <Text
          accessible={true}
          color="bluegrey"
          defaultColor="bluegrey"
          defaultWeight="Regular"
          font="TitilliumWeb"
          fontStyle={
            Object {
              "fontSize": 16,
              "lineHeight": 24,
            }
          }
          style={
            Array [
              Object {
                "textAlign": "center",
              },
              Object {
                "fontSize": 16,
                "lineHeight": 24,
              },
              Object {
                "color": "#475A6D",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "400",
              },
            ]
          }
          weight="Regular"
        >
          This is the loading caption
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        <View
          accessibilityLabel="Cancel"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
          testID="loadingSpinnerOverlayCancelButton"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 4,
                  "elevation": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingHorizontal": 16,
                  "textAlignVertical": "center",
                },
                false,
                Object {
                  "borderWidth": 1,
                },
                Object {
                  "height": 40,
                },
                false,
                Object {
                  "backgroundColor": "rgba(0,115,230,0)",
                  "borderColor": "#0073E6",
                },
                Object {
                  "backgroundColor": undefined,
                  "borderColor": undefined,
                  "transform": Array [
                    Object {
                      "scale": undefined,
                    },
                  ],
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              ellipsizeMode="tail"
              importantForAccessibility="no-hide-descendants"
              maxFontSizeMultiplier={1.3}
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "Titillium Web",
                    "fontStyle": "normal",
                    "fontWeight": "700",
                  },
                  Object {
                    "alignSelf": "center",
                  },
                  Object {
                    "color": "#0073E6",
                  },
                  Object {
                    "color": undefined,
                  },
                ]
              }
            >
              Cancel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "zIndex": 0,
        },
      ]
    }
  >
    <Text>
      This is a child
    </Text>
  </View>
</View>
`;

exports[`LoadingSpinnerOverlay Should match all-properties and not-loading snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
  testID="overlayComponent"
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "zIndex": 0,
        },
      ]
    }
  >
    <Text>
      This is a child
    </Text>
  </View>
</View>
`;

exports[`LoadingSpinnerOverlay Should match base loading snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
  testID="overlayComponent"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "bottom": 0,
          "inset": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1,
        },
        Object {
          "backgroundColor": "rgba(255,255,255,0.7)",
          "opacity": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": 100,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityRole="progressbar"
        accessible={true}
        importantForAccessibility="no-hide-descendants"
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
        testID="LoadingSpinnerTestID"
      >
        <View
          collapsable={false}
          style={
            Object {
              "transform": Array [
                Object {
                  "rotateZ": "0deg",
                },
              ],
            }
          }
          testID="LoadingSpinnerAnimatedTestID"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={24}
            bbWidth={24}
            fill="none"
            focusable={false}
            height={24}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={24}
          >
            <RNSVGGroup
              fill={null}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      736995,
                      1,
                      -16040221,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={0}
                  name="spinner-secondHalf"
                  x1="0%"
                  x2="100%"
                  y1="0%"
                  y2="0%"
                />
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -16040221,
                      1,
                      -16040221,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={0}
                  name="spinner-firstHalf"
                  x1="0%"
                  x2="100%"
                  y1="0%"
                  y2="0%"
                />
              </RNSVGDefs>
              <RNSVGGroup
                propList={
                  Array [
                    "strokeWidth",
                  ]
                }
                strokeWidth={3}
              >
                <RNSVGPath
                  d="M 1.5 12 A 10.5 10.5 0 0 1 22.5 12"
                  propList={
                    Array [
                      "stroke",
                    ]
                  }
                  stroke={
                    Array [
                      1,
                      "spinner-secondHalf",
                    ]
                  }
                />
                <RNSVGPath
                  d="M 22.5 12 A 10.5 10.5 0 0 1 1.5 12"
                  propList={
                    Array [
                      "stroke",
                    ]
                  }
                  stroke={
                    Array [
                      1,
                      "spinner-firstHalf",
                    ]
                  }
                />
                <RNSVGPath
                  d="M 1.5 12 A 10.5 10.5 0 0 1 1.5 11.25"
                  propList={
                    Array [
                      "stroke",
                      "strokeLinecap",
                    ]
                  }
                  stroke={4278927075}
                  strokeLinecap={1}
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
      <View
        style={
          Object {
            "padding": 24,
          }
        }
      >
        <Text
          accessible={true}
          color="bluegrey"
          defaultColor="bluegrey"
          defaultWeight="Regular"
          font="TitilliumWeb"
          fontStyle={
            Object {
              "fontSize": 16,
              "lineHeight": 24,
            }
          }
          style={
            Array [
              Object {
                "textAlign": "center",
              },
              Object {
                "fontSize": 16,
                "lineHeight": 24,
              },
              Object {
                "color": "#475A6D",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "400",
              },
            ]
          }
          weight="Regular"
        >
          Wait a few seconds..
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "zIndex": 0,
        },
      ]
    }
  />
</View>
`;

exports[`LoadingSpinnerOverlay Should match base no-loading snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
  testID="overlayComponent"
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "zIndex": 0,
        },
      ]
    }
  />
</View>
`;
